src/vs/workbench/services/bulkEdit/browser/bulkEditService.ts

```diff
@@ -410,7 +410,7 @@ export class BulkEditService implements IBulkEditService {
 			}
 		}
 
-		const bulkEdit = new BulkEdit(codeEditor, options.progress, this._logService, this._textModelService, this._fileService, this._textFileService, this._labelService, this._configurationService);
+		const bulkEdit = new BulkEdit(options.editor, options.progress, this._logService, this._textModelService, this._fileService, this._textFileService, this._labelService, this._configurationService);
```

Comment:
This is a huge change because it affects all users of bulk edit that rely on it, automatically trying to perform an edit using the editor. That's important so that selections and decorations stay stable whenever bulk edit affects a file that opened in an editor. 

If the "culprit" is the editor, checking that a model is readonly, (@isidorn please confirm, you have investigated this issue) then you wanna have a way of tell bulk edit to not use an editor. How about another property in `IBulkEditOptions` like `{ noFallbackEditor: boolean }` or using `{ editor: null }`?

Reply:
Yes I confirm. The culprit is the editor.